### PR TITLE
Fix blockdevice assert volume size

### PR DESCRIPTION
**Pull request by myechuri**
Round up size comparison in test_attach_volume_validate_size to GiB.

### CODE REVIEW

**Comment by wallrj** [comment]
@myechuri Is this for https://clusterhq.atlassian.net/browse/FLOC-1914 and is it up for review?

File: flocker/node/agents/ebs.py [hunk 1]
```diff
@@ -28,7 +28,7 @@
 METADATA_VERSION_LABEL = u'flocker-metadata-version'
 CLUSTER_ID_LABEL = u'flocker-cluster-id'
 ATTACHED_DEVICE_LABEL = u'attached-device-name'
-BOTO_NUM_RETRIES = u'10'
```
Comment (wallrj) [inline flocker/node/agents/ebs.py]: This change looks unrelated.

**Comment by wallrj** [comment]
So we'll be looking for the new rackspace storage driver builder to go green:
 * http://build.clusterhq.com/boxes-flocker?branch=fix-blockdevice-assert-volume-size

I forced a build.

**Comment by wallrj** [comment]
```
[FAIL]
Traceback (most recent call last):
  File "/srv/buildslave/flocker-functional-rackspace-centos-7-storage-driver/venv/lib/python2.7/site-packages/flocker/node/agents/test/test_blockdevice.py", line 1413, in test_attach_volume_validate_size
    self._assert_volume_size(attached_volume, REALISTIC_BLOCKDEVICE_SIZE)
  File "/srv/buildslave/flocker-functional-rackspace-centos-7-storage-driver/venv/lib/python2.7/site-packages/flocker/node/agents/test/test_blockdevice.py", line 1294, in _assert_volume_size
    self.assertEqual(device_size_GiB, volume_size_GiB)
  File "/srv/buildslave/flocker-functional-rackspace-centos-7-storage-driver/venv/lib/python2.7/site-packages/twisted/trial/_synctest.py", line 447, in assertEqual
    % (msg, pformat(first), pformat(second)))
twisted.trial.unittest.FailTest: not equal:
a = 100
b = 93


flocker.node.agents.functional.test_cinder.CinderBlockDeviceAPIInterfaceTests.test_attach_volume_validate_size
```

**Comment by wallrj** [comment]
I've done some testing on my Rackspace machine and it looks like they're interpreting the supplied size in `GiB` despite what the documentation says:
 * http://developer.openstack.org/api-ref-blockstorage-v2.html#createVolume

So I created a volume from inside a debug session and attached it to my host and got the following results:

```
(Pdb) nvol = self.api.create_volume(dataset_id=uuid4(), size=REALISTIC_BLOCKDEVICE_SIZE)    

(Pdb) nvol
BlockDeviceVolume(blockdevice_id=e1f26409-4019-42e5-a3e9-97e895fcf17c, dataset_id=c20fafc0-92a1-464f-9d55-174a50e08073, attached_to=None, size=100000000000)
```
I attached the new volume using the Rackspace API and waited for it to be listed as attached...


```
(Pdb) self.api.list_volumes()
[BlockDeviceVolume(blockdevice_id=e1f26409-4019-42e5-a3e9-97e895fcf17c, dataset_id=c20fafc0-92a1-464f-9d55-174a50e08073, attached_to=6ddfb6c0-d264-4e77-846a-aa67e4fe89df, size=100000000000), BlockDeviceVolume(blockdevice_id=68045749-d675-468b-aac8-7dec3e941722, dataset_id=97421826-ae5e-464c-a609-b24f3d3cfcdf, attached_to=6ddfb6c0-d264-4e77-846a-aa67e4fe89df, size=100000000000)]

``` 

Then got the device path from the API

```
(Pdb) self.api.get_device_path(nvol.blockdevice_id)
FilePath(u'/dev/xvdi')
```

And got the byte size using lsblk

```
~ $ lsblk  --bytes /dev/xvdi
NAME MAJ:MIN RM         SIZE RO TYPE MOUNTPOINT
xvdi 202:128  0 107374182400  0 disk 
```

Which corresponds to 100 GiB  not GB

```
(Pdb) Byte(107374182400).to_GB().value                                                                                                                                                                             
107.3741824
(Pdb) Byte(107374182400).to_GiB().value
100.0
```

So perhaps we need to change our cinder size conversion?

**Comment by wallrj** [comment]
Yeah, with the following modification:

```
(1914)~/.../ClusterHQ/flocker (fix-blockdevice-assert-volume-size ✕ =)$ git diff
diff --git a/flocker/node/agents/cinder.py b/flocker/node/agents/cinder.py
index 4c8e539..faca545 100644
--- a/flocker/node/agents/cinder.py
+++ b/flocker/node/agents/cinder.py
@@ -8,7 +8,7 @@ import time
 from uuid import UUID
 from subprocess import check_output
 
-from bitmath import Byte, GB
+from bitmath import Byte, GiB
 
 from eliot import Message, start_action
 
@@ -222,7 +222,7 @@ class CinderBlockDeviceAPI(object):
             # XXX: Address size mistach (see
             # (https://clusterhq.atlassian.net/browse/FLOC-1874).
             requested_volume = self.cinder_volume_manager.create(
-                size=Byte(size).to_GB().value,
+                size=Byte(size).to_GiB().value,
                 metadata=metadata,
             )
             Message.new(blockdevice_id=requested_volume.id).write()
@@ -374,7 +374,7 @@ def _blockdevicevolume_from_cinder_volume(cinder_volume):
 
     return BlockDeviceVolume(
         blockdevice_id=unicode(cinder_volume.id),
-        size=int(GB(cinder_volume.size).to_Byte().value),
+        size=int(GiB(cinder_volume.size).to_Byte().value),
         attached_to=server_id,
         dataset_id=UUID(cinder_volume.metadata[DATASET_ID_LABEL])
     )
diff --git a/flocker/testtools/__init__.py b/flocker/testtools/__init__.py
index 3c0c84d..1a0a5eb 100644
--- a/flocker/testtools/__init__.py
+++ b/flocker/testtools/__init__.py
@@ -21,7 +21,7 @@ from unittest import skipIf, skipUnless
 from inspect import getfile, getsourcelines
 from subprocess import PIPE, STDOUT, CalledProcessError, Popen
 
-from bitmath import GB
+from bitmath import GiB
 
 from pyrsistent import PRecord, field
 
@@ -55,7 +55,7 @@ from ..common.script import (
 # This is currently set to the minimum size for a SATA based Rackspace Cloud
 # Block Storage volume. See:
 # * http://www.rackspace.com/knowledge_center/product-faq/cloud-block-storage
-REALISTIC_BLOCKDEVICE_SIZE = int(GB(100).to_Byte().value)
+REALISTIC_BLOCKDEVICE_SIZE = int(GiB(100).to_Byte().value)
 
 
 @implementer(IProcessTransport)

```

The tests pass

```
(1914)~/.../ClusterHQ/flocker (fix-blockdevice-assert-volume-size ✕ =)$ sudo /usr/bin/env FLOCKER_FUNCTIONAL_TEST_OPENSTACK_REGION=DFW FLOCKER_FUNCTIONAL_TEST_CLOUD_PROVIDER=openstack FLOCKER_FUNCTIONAL_TEST_CLOUD_CONFIG_FILE=$HOME/acceptance.yml $(type -p trial) -b flocker.node.agents.functional.test_cinder.CinderBlockDeviceAPIInterfaceTests.test_attach_volume_validate_size
:0: UserWarning: You do not have a working installation of the service_identity module: 'No module named service_identity'.  Please install it from <https://pypi.python.org/pypi/service_identity> and make sure all of its dependencies are satisfied.  Without the service_identity module and a recent enough pyOpenSSL to support it, Twisted can perform only rudimentary TLS client hostname verification.  Many valid certificate/hostname mappings may be rejected.
> /home/richard/.virtualenvs/1914/lib/python2.7/site-packages/twisted/trial/runner.py(218)run()
-> try:
(Pdb) c
flocker.node.agents.functional.test_cinder
  CinderBlockDeviceAPIInterfaceTests
    test_attach_volume_validate_size ...                                   [OK]

-------------------------------------------------------------------------------
Ran 1 tests in 22.654s

PASSED (successes=1)
```

Does that seem plausible?

**Comment by wallrj** [comment]
http://bazaar.launchpad.net/~cinder-coresec/cinder/github/view/head:/cinder/volume/flows/api/create_volume.py#L35

```
GB = units.Gi
```

http://git.openstack.org/cgit/openstack/oslo.utils/tree/oslo_utils/units.py#n23

```
Gi = 1024 ** 3
```

**Comment by wallrj** [comment]
I created:
 * https://bugs.launchpad.net/openstack-api-site/+bug/1456631
 * https://mycloud.rackspace.com/cloud/929000//tickets#150519-ord-0001584

**Comment by exarkun** [comment]
Fixed this by fixing the actual Cinder backend instead - FLOC-1914 / #1427
